pocs/utils/logger.py

```diff
@@ -23,23 +50,41 @@ def get_root_logger(profile='panoptes', log_config=None):
         for name, formatter in log_config['formatters'].items():
             log_config['formatters'][name].setdefault('()', _UTCFormatter)
 
-    log_file_lookup = {
-        'all': "{}/logs/panoptes.log".format(os.getenv('PANDIR', '/var/panoptes')),
-        'warn': "{}/logs/warnings.log".format(os.getenv('PANDIR', '/var/panoptes')),
-    }
+        log_fname_datetime = datetime.datetime.utcnow().strftime('%Y%m%dT%H%M%SZ')
+    else:
+        log_fname_datetime = datetime.datetime.now().strftime('%Y%m%dT%H%M%SZ')
+
+    # Setup log file names
+    invoked_script = os.path.basename(sys.argv[0])
+    log_dir = '{}/logs'.format(os.getenv('PANDIR', gettempdir()))
+    log_fname = '{}-{}-{}'.format(invoked_script, os.getpid(), log_fname_datetime)
+    log_symlink = '{}/{}.log'.format(log_dir, invoked_script)
 
-    # Setup the TimeedRotatingFileHandler to backup in middle of day intead of middle of night
+    # Set log filename and rotation
     for handler in log_config.get('handlers', []):
-        log_config['handlers'][handler].setdefault('filename', log_file_lookup[handler])
-        if handler in ['all', 'warn']:
-            log_config['handlers'][handler].setdefault('atTime', datetime.time(hour=11, minute=30))
+        # Set the filename
+        full_log_fname = '{}/{}-{}.log'.format(log_dir, log_fname, handler)
+        log_config['handlers'][handler].setdefault('filename', full_log_fname)
+
+        # Setup the TimeedRotatingFileHandler for middle of day
+        log_config['handlers'][handler].setdefault('atTime', datetime.time(hour=11, minute=30))
+
+        if handler == 'all':
+            # Symlink the log file to $PANDIR/logs/panoptes.log
+            try:
+                os.unlink(log_symlink)
+            except FileNotFoundError:
+                pass
+            finally:
+                os.symlink(full_log_fname, log_symlink)
 
     # Configure the logger
     logging.config.dictConfig(log_config)
 
     # Get the logger and set as attribute to class
     logger = logging.getLogger(profile)
 
+    # Don't want log messages from state machine library
```

Comment:
I appreciate the comment as I had no idea what this was about. I wonder if this is the kind of thing that could be in conf_files/log.yaml. What do you think?